sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -308,7 +311,8 @@ def __init__(
       default_environment=None,
       bundle_repeat=0,
       use_state_iterables=False,
-      provision_info=None):
+      provision_info=None,
+      state_cache_size=100):
```

Comment:
Is this number of items? It'd probably be better for this to be sensitive to the size of the items (e.g. estimated by the serialized size we get when making state requests).

Reply:
This is the maximum number of items to hold in the cache. I agree it would be nice to have this size-based. Though this will also require a more sophisticated memory model to determine the correct memory size. We could go with something reasonable, e.g. 100mb or so.

Reply:
Yeah, 100mb sounds like a reasonable value. It also makes more sense as a good default (who knows what 100 items is) and something possibly a user could specify. 

I don't think the memory model would have to be that sophisticated; the serialized size should be a reasonable approximation. It would require tracking it though. 

I would be OK with a JIRA and a TODO here.

Reply:
Jira: https://issues.apache.org/jira/browse/BEAM-8297

Reply:
Also added in in the code.